### DISCUSSION TITLE
Usage without Session

### DISCUSSION THREAD

**Discussion by cs-miller** [TimelordUK/jspurefix#84] [Q&A]
Hi,
  Having a little difficulty determining if this library can be used to create/format/encode FIX messages without sending them/without establishing a session. Essentially I am sending FIX messages over MQ to a downstream system that will modify and send them.

**Comment by TimelordUK** [comment]
https://github.com/TimelordUK/jspurefix/blob/master/src/test/ascii/qf-50sp2-encoder.test.ts

if you look at this test file then class EncoderTest

this takes a message and encodes to a buffer without any physical session

**Reply by cs-miller** [reply]
Running into some issues following this approach because it is still using a session & the associated DI (if I understand correctly), which seems to fail because it is resolving the wrong path for data/dictionary.json.

I don't want to waste your time, as clearly the issue is I just don't understand the code/I'm using it in an unsupported way, but if there is a way I can initialize an encoder without a session & associated baggage (perhaps providing an absolute path to node_modules/jspurefix/data/dictionary.json ?) and go from JSON to ASCII let me know.

**Comment by TimelordUK** [comment]
if you look at https://github.com/TimelordUK/jspurefix/releases

there is a command line option using a simple wrapper script which could be used as a start point for example you can parse a fix log file into json messages.  This is not using a physical session its just setting up the engine and parsing

node node_modules/jspurefix/dist/jsfix-cmd --fix=../../jsfix.test_client.txt --delimiter="|"  --session=../../data/session/test-initiator.json --type=AE --objects


that may help as a start point maybe just try and run script and see output and then chop out the code you need - it is not very well written it started as a little test tool but it should be easy enough to adapt
